--- a/picom.sample.conf	2020-10-24 08:44:12.000000000 +0000
+++ b/picom.sample.conf	2021-01-23 07:25:31.706502061 +0000
@@ -7,8 +7,8 @@
 # (windows with '_NET_WM_WINDOW_TYPE_DESKTOP') never get shadow, 
 # unless explicitly requested using the wintypes option.
 #
-# shadow = false
-shadow = true;
+shadow = false
+# shadow = true;
 
 # The blur radius for shadows, in pixels. (defaults to 12)
 # shadow-radius = 12
@@ -87,8 +87,8 @@
 
 # Fade windows in/out when opening/closing and when opacity changes,
 #  unless no-fading-openclose is used.
-# fading = false
-fading = true
+fading = false
+# fading = true
 
 # Opacity change between steps while fading in. (0.01 - 1.0, defaults to 0.028)
 # fade-in-step = 0.028
@@ -117,12 +117,12 @@
 
 
 # Opacity of inactive windows. (0.1 - 1.0, defaults to 1.0)
-# inactive-opacity = 1
-inactive-opacity = 0.8;
+inactive-opacity = 1.0;
+# inactive-opacity = 0.8;
 
 # Opacity of window titlebars and borders. (0.1 - 1.0, disabled by default)
-# frame-opacity = 1.0
-frame-opacity = 0.7;
+frame-opacity = 1.0;
+# frame-opacity = 0.7;
 
 # Default opacity for dropdown menus and popup menus. (0.0 - 1.0, defaults to 1.0)
 # menu-opacity = 1.0
@@ -415,9 +415,9 @@
 #
 wintypes:
 {
-  tooltip = { fade = true; shadow = true; opacity = 0.75; focus = true; full-shadow = false; };
+  tooltip = { fade = false; shadow = false; opacity = 0.8; focus = true; full-shadow = false; };
   dock = { shadow = false; }
   dnd = { shadow = false; }
-  popup_menu = { opacity = 0.8; }
-  dropdown_menu = { opacity = 0.8; }
+  popup_menu = { opacity = 1.0; }
+  dropdown_menu = { opacity = 1.0; }
 };
