--- a/src/celestia/gtk/data/celestia.desktop	2019-10-05 00:32:23.000000000 +0300
+++ b/src/celestia/gtk/data/celestia.desktop	2020-01-18 20:49:03.269609300 +0300
@@ -2,12 +2,14 @@
 Encoding=UTF-8
 Version=1.0
 Type=Application
-Exec=celestia-gnome
+Exec=celestia
 Icon=celestia
 X-GNOME-DocPath=
 Terminal=false
 Name=Celestia (Gtk+)
 GenericName=Space Simulator
+GenericName[tr]=Uzay Simulatörü
 Comment=Explore the Universe in this detailed space simulation
+Comment[tr]=Evreni keşfetmek için 3B uzay simülatörü
 StartupNotify=true
 Categories=Astronomy;Education;Application;
