--- a/configure.ac	2021-02-20 14:47:26.805843365 +0000
+++ b/configure.ac	2021-02-20 14:52:50.227346104 +0000
@@ -12,17 +12,7 @@
 EXTRA_RELEASE=
 EXTRA_GTK2_VERSION=
 
-if test \( -d .git \); then
-    AC_CHECK_PROG([GIT], [git], [yes], [no], [$PATH])
-    if test \( "$GIT" = "no" \); then
-	AC_MSG_ERROR([*** git not found. See http://git-scm.com/])
-    else
-	GIT_VERSION=`git describe --abbrev=6 --dirty --always`
-	echo "echo ${GIT_VERSION}" > ./version
-    fi
-else
-    GIT_VERSION=`sh -c ". $srcdir/version"`
-fi
+GIT_VERSION=3.17.8-23-g1ebf74
 
 if test \( -z "$GIT_VERSION" \); then
 	AC_MSG_ERROR([*** could not determine program version])
