diff --git a/src/bin/e_main.c b/src/bin/e_main.c
index 55b96c2..ce73439 100644
--- a/src/bin/e_main.c
+++ b/src/bin/e_main.c
@@ -599,8 +599,8 @@ main(int argc, char **argv)
         /* who wish to use Qt apps without any gnome or gtk usage may choose to    */
         /* install qt5ct and overwrite this variable with qt5ct and use that to    */
         /* configure there Qt5 applications.                                       */
-        e_util_env_set("QT_QPA_PLATFORMTHEME", "gtk2");
-        e_util_env_set("QT_STYLE_OVERRIDE", "gtk2");
+        /* e_util_env_set("QT_QPA_PLATFORMTHEME", "gtk2");                         */
+        /* e_util_env_set("QT_STYLE_OVERRIDE", "gtk2");                            */
      }
    // make fonts NOT BLURRY. after 35 (v 38, v40 of interpreter) fonts become
    // horizontally blurry - they seemingly want the interpreter for hinting to
