From a8c3263dff3986b9c5ec1eb05b3acfb9e50970ac Mon Sep 17 00:00:00 2001
From: wb9688 <wb9688@users.noreply.github.com>
Date: Thu, 18 Mar 2021 12:03:55 +0100
Subject: [PATCH] Check if X11 before requesting X11 WM name

---
 src/main.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/main.c b/src/main.c
index a39faf1..aa6a8b0 100644
--- a/src/main.c
+++ b/src/main.c
@@ -157,8 +157,12 @@ extern gboolean plugin_load(LXAppearance* app, GtkBuilder* lxappearance_builder)
     if(app->abi_version > LXAPPEARANCE_ABI_VERSION)
         return FALSE;
 
+    GdkScreen *screen = gtk_widget_get_screen(app->dlg);
+    if(!GDK_IS_X11_SCREEN(screen))
+        return FALSE;
+
     /* detect openbox */
-    const char* wm_name = gdk_x11_screen_get_window_manager_name(gtk_widget_get_screen(app->dlg));
+    const char* wm_name = gdk_x11_screen_get_window_manager_name(screen);
     if(g_strcmp0(wm_name, "Openbox"))
         return FALSE; /* don't load the plugin if openbox is not in use. */
 
