diff --git a/gmusicbrowser.pl b/gmusicbrowser.pl
index 28f0489..72e25ee 100755
--- a/gmusicbrowser.pl
+++ b/gmusicbrowser.pl
@@ -7960,7 +7960,7 @@ sub ShowHide
 		{	next unless $win;
 			$win->{saved_position}=join 'x',$win->get_position;
 			$win->{skip_taskbar_hint}=$win->get_skip_taskbar_hint;
-			$win->set_skip_taskbar_hint(TRUE);
+			#$win->set_skip_taskbar_hint(TRUE);
 			$win->hide;
 		}
 	}
